src/lib/u2/browser.js

```diff
@@ -0,0 +1,571 @@
+      if ( index < 0 ) {
+        this.views.splice(0, this.views.length);
+        return;
+      }
+
+      while ( index + 1 < this.views.length ) {
+        this.removeTopmost_();
+      }
+    },
+
+    function addChild_(child) {
+      var index = this.views.length;
+      this.views.push(child);
+      this.viewAdded.pub(child, index);
+    },
+
+    // Should only be called on the topmost view.
+    function removeTopmost_() {
+      var e = this.views.pop();
+      this.viewRemoved.pub(e, this.views.length);
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'BasicStackView',
+  extends: 'foam.u2.View',
+  imports: [
+    'stack'
+  ],
+
+  properties: [
+    {
+      name: 'viewStack_',
+      factory: function() { return []; }
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this.stack.viewAdded.sub(this.onViewAdded);
+      this.stack.viewRemoved.sub(this.onViewRemoved);
+      if ( this.stack.views.length > 0 ) {
+        this.renderView_(this.stack.views[this.stack.views.length - 1]);
+      }
+    },
+    function renderView_(view) {
+      if ( this.viewStack_.length > 0 ) {
+        this.viewStack_[this.viewStack_.length - 1].hide();
+      }
+      this.viewStack_.push(view);
+      this.add(view);
+    },
+    function removeTopmost_() {
+      var e = this.viewStack_.pop();
+      e.remove();
+      if ( this.viewStack_.length > 0 ) {
+        this.viewStack_[this.viewStack_.length - 1].show(true);
+      }
+    }
+  ],
+
+  listeners: [
+    {
+      name: 'onViewAdded',
+      isFramed: true,
+      code: function onViewAdded(sub, _, view) {
+        this.renderView_(view);
+      }
+    },
+    {
+      name: 'onViewRemoved',
+      isFramed: true,
+      code: function onViewRemoved() {
+        this.removeTopmost_();
+      }
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'DAOUpdateController',
+  extends: 'foam.u2.View',
+  requires: [
+    'foam.u2.DetailView'
+  ],
+
+  imports: [
+    'document',
+    'stack'
+  ],
+
+  exports: [
+    'myControllerMode as controllerMode'
+  ],
+
+  properties: [
+    {
+      // The original, golden data. innerData is a clone of this.
+      name: 'data',
+      postSet: function(old, nu) {
+        // Only overwrite innerData if it's clean relative to old.
+        // That prevents overwriting any changes the user might have made.
+        if ( old && old.equals(this.innerData) ) {
+          this.innerData = nu.clone();
+        }
+      }
+    },
+    {
+      name: 'innerData',
+      factory: function() {
+        return this.data.clone();
+      },
+      postSet: function(old, nu) {
+        if ( old === nu ) return;
+        if ( this.dataSub_ ) this.dataSub_.destroy();
+        this.dataSub_ = nu.propertyChange.sub(this.onDataChange);
+        this.onDataChange();
+      }
+    },
+    'dataSub_',
+    {
+      name: 'of',
+      expression: function(data) {
+        return data.cls_;
+      }
+    },
+    {
+      name: 'dao',
```

Comment:
I've added `foam.String.daoize()` that builds the properly-lowercased `someClassDAO`.

(With tests, naturally.)